### DISCUSSION TITLE
Windows applications support both x86 and x64？

### DISCUSSION THREAD

**Discussion by RayGyoe** [airsdk/Adobe-Runtime-Support#1592] [Ideas]
Ane supports multiple architectures, but publishing applications is only for specific architectures. Can I have two architecture runtimes at the same time in the future?

**Comment by ajwfrost** [comment]
Hi

This I think would depend on the packaging format supported by a platform. So e.g. on macOS we generate 'fat'/universal binaries that contain both an x86_64 and an arm64 block within a single file; on Android there's a bundle format that will contain x86/x64/armv7/armv8 binaries all together (although when it goes onto the device, it's stripped by the delivery mechanism into just whatever the device supports...)

For Windows ... I'm not sure that such a mechanism exists? If there is please let me know, I just did a quick google search but may not be looking for exactly the right thing.

The ".air" format is agnostic and then is just installed based on whatever runtime the end user has on their computer (which invariably would be 32-bit at the moment, because that's the only type of AIR shared runtime has been released afaik - although we are planning to change that!).

And if you're packaging up for bundles/native installers, you can use the "-arch" option to switch over from 32-bit to 64-bit. Not sure whether this is particularly well supported by the IDEs though..? 

So - hope that's a little helpful, but I don't think it's possible to have a single application support multiple cpu architectures on Windows; happy to be corrected though!

thanks
